field/TinyLang/Field/Generator.hs

```diff
@@ -106,23 +103,34 @@ adjustUniquesForVars :: MonadSupply m => Vars f -> m ()
 adjustUniquesForVars =
     supplyFromAtLeast . freeUniqueFoldable . map (forget $ _varUniq . _uniVarVar)
 
--- | The variables used by our generators by default.
-defaultVars :: Vars f
-defaultVars = runSupply $ do
-    let make uni name = Some . UniVar uni <$> freshVar name
-    fieldVars  <- traverse (make Field ) ["x", "y", "z", "p", "q", "r", "s", "t"]
-    boolVars   <- traverse (make Bool  ) ["?a", "?b", "?c", "?d", "?e", "?f", "?g", "?h"]
-    vectorVars <- traverse (make Vector) ["#q", "#r", "#s", "#t", "#u", "#v", "#w"]
-    return $ fieldVars ++ boolVars ++ vectorVars
 
 -- | A wrapper around @UniVar f a@ provided for its @Arbitrary@ instance that allows to generate
 -- variables from the default set of them.
-newtype DefaultUniVar f a = DefaultUniVar
-    { unDefaultUniVar :: UniVar f a
+
+defaultIdents :: [String]
+defaultIdents = concat . transpose $ [fieldIdents, boolIdents, vectorIdents] where
+        fieldIdents   = ["x", "y", "z", "p", "q", "r", "s", "t"]
+        boolIdents    = ["?a", "?b", "?c", "?d", "?e", "?f", "?g", "?h"]
+        vectorIdents  = ["#q", "#r", "#s", "#t", "#u", "#v", "#w"]
+
+newtype Default a = Default
+    { unDefault :: a
     }
+    deriving (Show, Eq)
+
+-- type DefaultUniVar f a    = Default (UniVar f a)
+type DefaultVars          = Default [Var]
+type DefaultSomeUniVars f = Default (Vars f)
+
+instance Arbitrary DefaultVars where
+    arbitrary = do
+        vars <- sublistOf $ defaultIdents
+        pure . Default $ runSupply $ traverse freshVar vars
+
+instance Arbitrary (DefaultSomeUniVars f) where
+    arbitrary = Default . fmap mkSomeUniVar . unDefault <$> arbitrary
```

Comment:
No, because `traverse` will collect the side effect of `[]` and we will end up with `[Default a]` not `Default [a]`.

If we want to play code golf, it is:
``` haskell
arbitrary = (fmap . fmap . fmap) mkSomeUniVar arbitrary
```

Reply:
Ah, yeah, `mkSomeUniVar` is not monadic. But then 

```haskell
arbitrary = fmap mkSomeUniVar <$> arbitrary
```

should work?

```haskell
arbitrary = (fmap . fmap . fmap) mkSomeUniVar arbitrary
```

Do you need that many `fmap`s?